draft-js-resizeable-plugin/src/createDecorator.js

```diff
@@ -87,24 +87,38 @@ export default ({ config, store }) => (WrappedComponent) => class BlockResizeabl
 
     // Do the actual drag operation
     const doDrag = (dragEvent) => {
-      let width = (startWidth + dragEvent.clientX) - startX;
+      let width = startWidth + (isLeft ? startX - dragEvent.clientX : dragEvent.clientX - startX);
       let height = (startHeight + dragEvent.clientY) - startY;
 
       const editorComp = store.getEditorRef();
       // this keeps backwards-compatibility with react 15
       const editorNode = editorComp.refs.editor ? editorComp.refs.editor : editorComp.editor;
 
-      width = editorNode.clientWidth < width ? editorNode.clientWidth : width;
-      height = editorNode.clientHeight < height ? editorNode.clientHeight : height;
+      width = Math.min(editorNode.clientWidth, width);
+      height = Math.min(editorNode.clientHeight, height);
 
       const widthPerc = (100 / editorNode.clientWidth) * width;
       const heightPerc = (100 / editorNode.clientHeight) * height;
 
       const newState = {};
-      if ((isLeft || isRight) && horizontal === 'relative') {
-        newState.width = resizeSteps ? round(widthPerc, resizeSteps) : widthPerc;
-      } else if ((isLeft || isRight) && horizontal === 'absolute') {
-        newState.width = resizeSteps ? round(width, resizeSteps) : width;
+      switch (true) {
```

Comment:
that's a very unconventional use of the switch statement, mind sticking to if statements for this block?

Reply:
Ah, I just realized that part of switch code is useless >_<, , i'll undo it in next few minutes